src/metabase/models/serialization.clj

```diff
@@ -0,0 +1,981 @@
+(ns metabase.models.serialization
```

Comment:
To save me from diffing three files against this with my eyeballs: is this the old `metabase.models.serialization.{base,hash,util}` poured into one file with minimal changes (eg. renaming `serdes-descendants` to `descendants`)?

Reply:
I notice it's substantially shorter, perhaps by 300LOC. Is that all from dropping the ingestion paths, or has something else been removed?

Reply:
Yes, it's a combo of the old base/hash/util namespaces, since there weren't any circular dependencies between them or the other namespaces that used them. With more functions in one place, it's easier to spot patterns that could be abstracted away. Some functions have been rewritten or replaced. Most names are shorter, and I have a few more ideas with some of the multimethods that I'll get to later.

Some big chunks of comments/docstrings are gone since the code they described has changed substantially, and I'm planning on adding docs where needed while I continue on serdes issues.

I'll add some comments highlighting changes.